Feat locales info options (#8985)
mdn/content: files/en-us/web/javascript/reference/global_objects/intl/locale/timezones/index.md

---
title: Intl.Locale.prototype.timeZones
slug: Web/JavaScript/Reference/Global_Objects/Intl/Locale/timeZones
tags:
  - Internationalization
  - Intl
  - JavaScript
  - Property
  - Locale
  - Localization
  - Prototype
  - Reference
  - timeZones
  - time zone
browser-compat: javascript.builtins.Intl.Locale.timeZones
---
{{JSRef}}

> The **`Intl.Locale.prototype.timeZones`** property is an accessor property which returns a `List` of supported time zones for choosen `Locale`.

Reviewer: ```suggestion
The **`Intl.Locale.prototype.timeZones`** property is an accessor property which returns an array of supported time zones for a chosen `Locale`.
```

## Description

Returns an array with supported time zones for the associated `Locale`, returned time zones
represents an [IANA time zone](https://en.wikipedia.org/wiki/Daylight_saving_time#IANA_time_zone_database).

> **Note:** If the Unicode Language Identifier does not contain the the `-` for the Unicode region subtag sequence, the returned value is `undefined`

## Examples

### Obtaining supported time zones

List supported time zones for a given `Locale`.

```js
let arEG = new Intl.Locale("ar-EG");

> console.log(arEG.timeZones); // Prints ["Africa/Cairo"]

Reviewer: ```suggestion
console.log(arEG.timeZones); // logs ["Africa/Cairo"]
```

```

```js
let jaJP = new Intl.Locale("ja-JP");

> console.log(jaJP.hourCycles); // Prints ["Asia/Tokyo"]

Reviewer: ```suggestion
console.log(jaJP.hourCycles); // logs ["Asia/Tokyo"]
```

```

```js
let ar = new Intl.Locale("ar");

> console.log(ar.hourCycles); // Prints undefined

Reviewer: ```suggestion
console.log(ar.hourCycles); // logs undefined
```
